s3po/connection.py

```diff
@@ -1,142 +1,59 @@
-'''Connecting to and talking to S3'''
+'''Deal with object storage.'''
 
+import contextlib
 import os
-import warnings
 from cStringIO import StringIO
-from boto.s3.connection import S3Connection
 
 # Internal imports
-from .mock import Mock
-from .util import retry
-from .util import CountFile, logger
-from .exceptions import UploadException, DownloadException
+from .util import retry, logger
+from .backends.s3 import S3
+from .backends.swift import Swift
+from .backends.memory import Memory
 
 
 class Connection(object):
     '''Our connection to S3'''
-    # How big must a file get before it's multiparted. Also how big the chunks
-    # are that we'll read
-    multipart_chunk = 50 * 1024 * 1204
 
-    def __init__(self, *args, **kwargs):
-        self.conn = S3Connection(*args, **kwargs)
+    @classmethod
+    def s3(cls, *args, **kwargs):  # pragma: no cover
```

Comment:
It seems like you could at least test the creation of a connection without doing any actual upload/download to it. That's a marginal increase in coverage.